SDWebImage/SDWebImageDownloaderOperation.m

```diff
@@ -301,6 +301,10 @@ - (void)URLSession:(NSURLSession *)session
             progressBlock(0, expected, self.request.URL);
         }
     } else {
+        [self callCompletionBlocksWithError:[NSError errorWithDomain:SDWebImageServerErrorDomain code:statusCode userInfo:@{NSLocalizedDescriptionKey : [NSString stringWithFormat:@"Server errors reported, the HTTP response code is %ld", (long)statusCode]}]];
+        LOCK(self.callbacksLock);
```

Comment:
Previously, I think maybe we can call `done` method directly, but I don't ensure wether it would leads some issues when `invalidateAndCancel` `ownedSession` in session delegate method.